packages/wonder-blocks-search-field/src/components/search-field.js

```diff
@@ -0,0 +1,257 @@
+    ...AriaProps,
+
+    /**
+     * ARIA label for the clear button. Defaults to "Clear search".
+     */
+    clearAriaLabel?: string,
+
+    /**
+     * The unique identifier for the input.
+     */
+    id: string,
+
+    /**
+     * The text input value.
+     */
+    value: string,
+
+    /**
+     * Provide hints or examples of what to enter. This shows up as
+     * a grayed out text in the field before a value is entered.
+     */
+    placeholder?: string,
+
+    /**
+     * Makes a read-only input field that cannot be focused.
+     * Defaults to false.
+     */
+    disabled?: boolean,
+
+    /**
+     * Changes the default focus ring color to fit a dark background.
+     */
+    light?: boolean,
+
+    /**
+     * Custom styles for the main wrapper.
+     */
+    style?: StyleType,
+
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string,
+
+    /**
+     * Called when the value has changed.
+     */
+    onChange: (newValue: string) => mixed,
+
+    /**
+     * Handler that is triggered when this component is clicked. For example,
+     * use this to adjust focus in parent component. This gets called when we
+     * click the dismiss icon button within the SearchField.
+     */
+    onClick?: () => mixed,
+
+    /**
+     * Called when a key is pressed.
+     */
+    onKeyDown?: (event: SyntheticKeyboardEvent<HTMLInputElement>) => mixed,
+
+    /**
+     * Called when the element has been focused.
+     */
+    onFocus?: (event: SyntheticFocusEvent<HTMLInputElement>) => mixed,
+
+    /**
+     * Called when the element has been blurred.
+     */
+    onBlur?: (event: SyntheticFocusEvent<HTMLInputElement>) => mixed,
+|};
+
+type DefaultProps = {|
+    clearAriaLabel: $PropertyType<Props, "clearAriaLabel">,
+    disabled: $PropertyType<Props, "disabled">,
+    light: $PropertyType<Props, "light">,
+|};
+
+type State = {|
+    /**
+     * The user focuses on this field.
+     */
+    focused: boolean,
+|};
+
+export default class SearchField extends React.Component<Props, State> {
+    static defaultProps: DefaultProps = {
+        clearAriaLabel: defaultLabels.clearSearch,
+        disabled: false,
+        light: false,
+    };
+
+    state: State = {
+        focused: false,
+    };
+
+    handleDismiss: () => void = () => {
+        const {onClick, onChange} = this.props;
+        // Empty the search text and focus the SearchField
+        onChange("");
+        if (onClick) {
+            onClick();
+        }
+    };
+
+    maybeRenderClearIconButton(): React.Node {
+        const {clearAriaLabel, value} = this.props;
+
+        if (value.length > 0) {
+            return (
+                <IconButton
+                    icon={icons.dismiss}
+                    kind="tertiary"
+                    onClick={this.handleDismiss}
+                    style={styles.dismissIcon}
+                    aria-label={clearAriaLabel}
+                />
+            );
+        }
+        return null;
```

Comment:
Nit: This works perfectly, but you could also do the opposite... check for null inside the if, otherwise return the `IconButton` instance. Any option works, this just improves indentation a bit (so no worries if you keep it as it is).